server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -26,17 +27,17 @@ func (api *API) InitOutgoingOAuthConnection() {
 	api.BaseRoutes.OutgoingOAuthConnection.Handle("", api.APISessionRequired(getOutgoingOAuthConnection)).Methods("GET")
 	api.BaseRoutes.OutgoingOAuthConnection.Handle("", api.APISessionRequired(updateOutgoingOAuthConnection)).Methods("PUT")
 	api.BaseRoutes.OutgoingOAuthConnection.Handle("", api.APISessionRequired(deleteOutgoingOAuthConnection)).Methods("DELETE")
-	api.BaseRoutes.OutgoingOAuthConnection.Handle("/validate", api.APISessionRequired(validateOutgoingOAuthConnectionCredentials)).Methods("POST")
 	api.BaseRoutes.OutgoingOAuthConnections.Handle("/validate", api.APISessionRequired(validateOutgoingOAuthConnectionCredentials)).Methods("POST")
 }
 
 // checkOutgoingOAuthConnectionReadPermissions checks if the user has the permissions to read outgoing oauth connections.
-// The user needs to have the permissions to manage outgoing webhooks and slash commands in order to read outgoing oauth connections
-// so that they can use them in their outgoing webhooks and slash commands.
+// An user with the permissions to manage outgoing oauth connections can read outgoing oauth connections.
+// Otherwise the user needs to have the permissions to manage outgoing webhooks and slash commands in order to read outgoing
+// oauth connections so that they can use them in their outgoing webhooks and slash commands.
 // This is made in this way so only users with the management permission can setup the outgoing oauth connections and then
 // other users can use them in their outgoing webhooks and slash commands if they have permissions to manage those.
 func checkOutgoingOAuthConnectionReadPermissions(c *Context) bool {
-	if c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageOutgoingWebhooks) && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSlashCommands) {
+	if c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageOutgoingOAuthConnections) || (c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageOutgoingWebhooks) && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSlashCommands)) {
```

Comment:
Thinking more about the read permissions:

The `PermissionManageOutgoingOAuthConnections` makes sense. 

But why would we otherwise require **both** `PermissionManageOutgoingWebhooks` **and** `PermissionManageSlashCommands`. We could have a case where e.g. Team admins are granted slash commands but not outgoing webhooks permissions so that they can create slash commands. Slash commands don't require outgoing webhooks. Why would just the `PermissionManageSlashCommands` not be sufficient from granting them read access to the outgoing connections?

Reply:
You are right. For read, you should only need one of the two, and even then, should we trim the output of the API? Since the permissions are intended only for usage, not for retrieving data.

Reply:
Yes that should be the ideal. Do we need anything more than the name for the slash commands and the outgoing webhooks?

Reply:
So with `PermissionManageOutgoingOAuthConnections` we don't need to trim, but with either `PermissionManageSlashCommands` or `PermissionManageOutgoingWebhooks`, we should respond only with the data needed